pkg/http/oidc_authenticator_test.go

```diff
@@ -55,26 +55,47 @@ func TestOIDCAuthenticator(t *testing.T) {
 	roundTripper := mock.NewMockRoundTripper(ctrl)
 	randomNumberGenerator := mock.NewMockThreadSafeGenerator(ctrl)
 	cookieAEAD := mock.NewMockAEAD(ctrl)
-	cookieAEAD.EXPECT().NonceSize().Return(4)
-	authenticator, err := bb_http.NewOIDCAuthenticator(
-		&oauth2.Config{
-			ClientID:     "MyClientID",
-			ClientSecret: "MyClientSecret",
-			Endpoint: oauth2.Endpoint{
-				AuthURL:   "https://login.com/authorize",
-				TokenURL:  "https://login.com/token",
-				AuthStyle: oauth2.AuthStyleInParams,
-			},
-			RedirectURL: "https://myserver.com/callback",
-			Scopes:      []string{"openid", "email"},
+	oauth2Config := &oauth2.Config{
+		ClientID:     "MyClientID",
+		ClientSecret: "MyClientSecret",
+		Endpoint: oauth2.Endpoint{
+			AuthURL:   "https://login.com/authorize",
+			TokenURL:  "https://login.com/token",
+			AuthStyle: oauth2.AuthStyleInParams,
 		},
+		RedirectURL: "https://myserver.com/callback",
+		Scopes:      []string{"openid", "email"},
+	}
+
+	cookieAEAD.EXPECT().NonceSize().Return(4)
+	userInfoClaimsFetcher := bb_http.NewUserInfoOIDCClaimsFetcher(
+		oauth2Config,
 		"https://login.com/userinfo",
+	)
+	userInfoAuthenticator, err := bb_http.NewOIDCAuthenticator(
+		oauth2Config,
+		userInfoClaimsFetcher,
+		jmespath.MustCompile("{\"public\": @}"),
+		&http.Client{Transport: roundTripper},
+		randomNumberGenerator,
+		"CookieName",
+		cookieAEAD,
+		clock,
+	)
+
+	cookieAEAD.EXPECT().NonceSize().Return(4)
+	idTokenClaimsFetcher := bb_http.NewIDTokenOIDCClaimsFetcher()
+	idTokenAuthenticator, err := bb_http.NewOIDCAuthenticator(
+		oauth2Config,
+		idTokenClaimsFetcher,
 		jmespath.MustCompile("{\"public\": @}"),
 		&http.Client{Transport: roundTripper},
 		randomNumberGenerator,
 		"CookieName",
 		cookieAEAD,
-		clock)
+		clock,
+	)
```

Comment:
Maybe readability of the tests would increase if you added additional `t.Run()` blocks here?

```go
func TestOIDCAuthenticator(t *testing.T) {
    t.Run("UserInfo", func(t *testing.T) { /* all of the existing tests go here */ })
    t.Run("IDToken", func(t *testing.T) { /* some tests go here */ })
}
```

Then you can also keep the name `authenticator` instead of renaming it to `userInfoAuthenticator`.